R/archivePackages.R

```diff
@@ -1,93 +1,128 @@
 
-##' The function moves older versions of packages into a CRAN-style
+DRAT_ARCHIVE_SUB_DIR <- "Archive"
+
+DRAT_BINARY_TYPES <- c("mac.binary","mac.binary.el-capitan",
+                       "mac.binary.mavericks", "win.binary")
+DRAT_BOTH_TYPES <- c("source", "mac.binary","mac.binary.el-capitan",
+                     "mac.binary.mavericks", "win.binary")
+
+DRAT_VERSION_REGEX <- "[0-9]\\.[0-9]$"
+DRAT_CONTRIB_VERSION_REGEX <- paste0("contrib/",DRAT_VERSION_REGEX)
+
+.norm_type <- function(type){
+    type <- match.arg(type,c("source", "binary", "mac.binary","mac.binary.el-capitan",
+                             "mac.binary.mavericks", "win.binary", "both"), 
+                      several.ok = TRUE)
+    if("both" %in% type){
+        return(DRAT_BOTH_TYPES)
+    }
+    if("binary" %in% type){
+        type <- c(type,DRAT_BINARY_TYPES)
+        type <- unique(type)
+    }
+    type
+}
+
+.norm_pkg <- function(pkg){
+    if(missing(pkg)){
+        pkg <- NULL
+    } else {
+        pkg <- unique(pkg)
+    }
+    pkg
+}
+
+##' The function moves older versions of packages into a CRAN-style 
 ##' archive folder.
-##'
+##' 
```

Comment:
Seeing one commit to remove empty trailing space, and soon after another commit to introduce it is kind of funny.
I suggest to always check `git diff --cached` before doing `commit` to ensure that lines which are affected are the lines necessary for logic change. If one wants to change existing style/formattng, then it is always better to put into separate commit, or even better, not do it at all. The smaller the diff, the better, and easier to review. This commit did actually very well by removing previously affected lines.

Reply:
Not all editors are equally smart. Adding `.editorconfig` helps but not all editors adhere to it...

Reply:
I maybe added that. I know it says outdated but I had to add the `@name` tag and RStudio behaves a bit differently, when automatically adding new lines in the comment section.